z3/snap.py

```diff
@@ -573,17 +573,15 @@ def main():
     cfg = get_config()
     args = parse_args()
 
-    try:
+    extra_config = {}
+    if 'HOST' in cfg:
+        extra_config['host'] = cfg['HOST']
+    if 'S3_KEY_ID' in cfg:
         s3_key_id, s3_secret, bucket = cfg['S3_KEY_ID'], cfg['S3_SECRET'], cfg['BUCKET']
-
-        extra_config = {}
-        if 'HOST' in cfg:
-            extra_config['host'] = cfg['HOST']
-    except KeyError as err:
-        sys.stderr.write("Configuration error! {} is not set.\n".format(err))
-        sys.exit(1)
-
-    bucket = boto.connect_s3(s3_key_id, s3_secret, **extra_config).get_bucket(bucket)
+        bucket = boto.connect_s3(s3_key_id, s3_secret, **extra_config).get_bucket(bucket)
+    else:
+        bucket = cfg['BUCKET']
+        bucket = boto.connect_s3( **extra_config).get_bucket(bucket)
```

Comment:
There is an extra ` `.

Reply:
I don't see much use of host parameter for accessing s3. I assume AWS will choose the best path. Also, is region a better option when comparing with static endpoint?

Reply:
I agree extra config is useful. I will test it further and commit it again

Reply:
add extra_config support, tested fine

Reply:
👍